src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQuery.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryVisitor;
+import org.apache.lucene.search.ScoreMode;
+import org.apache.lucene.search.Weight;
+
+/**
+ * Implementation of Query interface for type NeuralSparseQuery when TwoPhaseNeuralSparse Enabled.
+ * Initialized, it currentQuery include all tokenQuery. After call setCurrentQueryToHighScoreTokenQuery,
+ * it will perform highScoreTokenQuery.
+ */
+@AllArgsConstructor
+@Getter
+@NonNull
+public final class NeuralSparseQuery extends Query {
+
+    private Query currentQuery;
+    private final Query highScoreTokenQuery;
+    private final Query lowScoreTokenQuery;
+    private final Float rescoreWindowSizeExpansion;
+
+    /**
+     *
+     * @param field The field of this query.
+     * @return String of NeuralSparseQuery object.
+     */
+    @Override
+    public String toString(String field) {
+        return "NeuralSparseQuery("
```

Comment:
can we use String.format or StringBuilder for string concatenation?

Reply:
Replaced with Format.